### DISCUSSION TITLE
DOI problem

### DISCUSSION THREAD

**Discussion by ludowaltman** [knowledgefutures/pubpub#2209] [Support]
On August 12 I published a pub (https://ludowaltman.pubpub.org/pub/review-ikf/release/1) and submitted the metadata to Crossref. However, the DOI assigned to my pub (https://doi.org/10.21428/7ccec04a.ace5807f) doesn't work, even after waiting for more than two weeks and resubmitting the metadata to Crossref several times. I didn't have this problem with earlier pubs.

My problem seems similar to the following problem discussed two months ago: https://github.com/pubpub/pubpub/discussions/2079.

Any help would be appreciated. Thanks!

Ludo

**Comment by gabestein** [comment]
Hey Ludo — feel free to write in sooner next time this happens. It should take 24 hours max, not two weeks! In this case, behind the scenes Crossref is complaining that the DOI of the connected arxiv preprint (10.48550/arXiv.2207.10722) isn't a valid DOI. Which is weird, because it is valid as far as I can tell.

I think it may be an issue on their end, and I'll write to them to see what's up. As a workaround for now, you could try adding the connection via the full arxiv URI rather than the DOI and depositing that way.

_As an aside, I really wish there was a way to display these errors in real-time, but unfortunately their system doesn't work that way._

**Reply by gabestein** [reply]
Heard back from Crossref. Unfortunately, this is the response: 

> The peer review content type does not (yet) allow relationships between DOIs registered by other registration agencies. I've been told this was a limitation of our current billing code, so not likely a quick fix. Unfortunately, you'll have to omit the relationship metadata from this registration in order to register the DOI with us.

So, I think you'll need to use the URI for the connection rather than the DOI, or omit the connection and deposit it, then re-add the connection after it's been deposited. I will, of course, make the case that they should prioritize fixing this — in particular, it's non-trivial to detect, from a DOI, whether it's registered by Crossref or someone else, so it's hard to prevent this error from occurring, and then of course hard to tell people about it when it does because of the way the queue works.

**Reply by gabestein** [reply]
Crossref has opened a request to add this support. They've cautioned it may take a while since it involves their billing code, but I do think it's worth chiming in if you'd like to see this feature implemented — and personally, I do think it's important for the future of preprint review.

https://crossref.atlassian.net/browse/CR-777

**Comment by ludowaltman** [comment]
Hi Gabe, thanks so much for your help! I just published a new review on PubPub. It's again a review of an arXiv preprint, but this time I used the arXiv URI rather than the arXiv DOI to create a connection from my review to the preprint. However, there again seems to be a problem with the DOI of my review: https://doi.org/10.21428/7ccec04a.4953b267. It doesn't work. It's supposed to refer to https://ludowaltman.pubpub.org/pub/review-jif-pr/release/1. Is this again a Crossref issue?

**Reply by gabestein** [reply]
Sorry about that Ludo. Looks like we're still getting the same error on the uri, which is surprising. I've submitted another ticket and will follow up when I hear back.

**Reply by gabestein** [reply]
Well, here's the unfortunate (but very understanding) response from Crossref. Long story short, it turns out they currently don't support depositing peer reviews on _any_ non-Crossref-DOI content. That's obviously hugely disappointing and a big blocker for a lot of our work. I think we're going to need to make a little more noise around this as a community if we want to see some action in the near future.

> Unfortunately, the only relationships that can currently be established for the peer review content type are relations to other Crossref DOIs. Our product team is discussing whether or not we'll be expanding relationships for peer reviews beyond Crossref DOIs. Several of us have had an initial discussion about it, but I don't see us reaching a decision quickly on this and certainly don't think we'll be expanding these relationships in the shorter term. We'll follow up on https://crossref.atlassian.net/browse/CR-777 as we have more thoughts/actions.

I would recommend posting a brief (kind) note in support of prioritizing this feature more highly here: https://crossref.atlassian.net/browse/CR-777, and asking others to do the same.

As a workaround, you can remove the connection, deposit the DOI, and then add it back. That will at least get you a DOI, even though it won't have the relationship to the arxiv preprint.

I should also note we're working on DataCite support as we speak. I'm not 100% sure if they support cross-DOI-depositor peer review relationships, but their schema indicates they do. If they do, we will recommend folks who need it switch as soon as we have that available.

**Reply by gabestein** [reply]
One (extremely silly) hack/workaround I just thought of and you probably shouldn't do, but that demonstrates the ridiculousness of the lack of support: I _think_ you could create a Pub with an `isVersionOf` Connection to the arxiv preprint, give that a DOI, and then deposit a review with that Pub you just created as the `isReviewOf` connection.
